MiniApp/Classes/MiniAppDownloader.swift

```diff
@@ -1,16 +1,12 @@
-/**
- * Protocol for the Downloader class that handles downloading the files for Mini App.
- */
-protocol MiniAppDownloaderProtocol {
+class MiniAppDownloader {
 
-    /// Handles downloading each file to a specific location.
-    /// - Parameters:
-    ///   - urls: [String] - List of URLs in the manifest.
-    ///   - pathDirectory: URL - The system path directory to cache.
-    func download(_ urls: [String], to pathDirectory: URL)
-}
+    typealias DownloadCompletionHandler = (Result<URL?, Error>) -> Void
 
-class MiniAppDownloader: NSObject, MiniAppDownloaderProtocol {
+    var miniAppClient: MiniAppClient?
+    var miniAppStorage: MiniAppStorage
+    var manifestDownloader: ManifestDownloader
+    var completionHandler: DownloadCompletionHandler?
+    var miniAppDirectory: URL?
```

Comment:
Why not make these dependencies private and inject them in init?

Reply:
Done. Couldn't move everything, but moved few that can be injected. Thanks :)